### DISCUSSION TITLE
srsRAN gNB with COTS UEs setup using USRP X410

### DISCUSSION THREAD

**Discussion by fahad-ece** [srsran/srsRAN_Project#462] [General Help]
Hello Everyone,

I would like to implement the srsRAN gNB with COTS UEs using USRP X-410. Unfortunately there is no srsRAN configuration file for X410. For srsRAN configuration- I was trying to follow the files from the following website: https://github.com/srsran/srsRAN_Project/blob/main/configs/gnb_rf_b200_tdd_n78_20mhz.yml . Can anyone please guide me with the necessary modification to complete this setup?

**Comment by zhuangyufan** [comment]
The following configuration works well for X410.

amf:
  addr: 192.168.18.196
  bind_addr: 192.168.18.197  

ru_sdr:
  device_driver: uhd       
  device_args: type=x4xx,num_recv_frames=64,num_send_frames=64 
  srate: 122.88                                                   

  tx_gain: 50                                                    
  rx_gain: 40                                                     

cell_cfg:
  dl_arfcn: 632628                                               
  band: 78                                                      
  channel_bandwidth_MHz: 100                                      
  common_scs: 30                                                
  plmn: "30030"                                                 
  tac: 100                                                    
  pci: 1                                                  

log:
  filename: ./gnb.log                                     
  all_level: info                                          

pcap:
  mac_enable: false                                             
  mac_filename: /tmp/gnb_mac.pcap                           
  ngap_enable: false                                             
  ngap_filename: /tmp/gnb_ngap.pcap

**Reply by jarenass001** [reply]
Did you use any iptables rule or any additional nat rule to redirect traffic when using x410? 
I am implementing it and the UE messages are not getting through to the gnb and i suspect that is the problem, because I used a spectrum analyser and x410 can transmit well.
I did a COTS UE implementation using b210 and all worked well. I don't know if there are some different steps to follow or something else.
